gulpTasks/doc.js

```diff
@@ -24,40 +32,66 @@ gulp.task('doc', ['copyBinToDoc', 'buildPlayground'], function() {
   const src = app.expandInputFiles(['src']);
   const project = app.convert(src);
   app.generateDocs(project, 'docgen');
-  app.generateJson(project, './bin/docgen/docgen.json', 'https://coveo.github.io/search-ui/');
+  app.generateJson(project, docgenJsonPath, 'https://coveo.github.io/search-ui/');
   return gulp.src('./readme.png').pipe(gulp.dest('./docgen'));
 });
 
 gulp.task('cleanGeneratedThemesFiles', () => {
   return del(['./docs/theme/assets/gen/**/*']);
 });
 
-gulp.task('copyBinToDoc', ['cleanGeneratedThemesFiles'], function() {
+gulp.task('copyBinToDoc', ['cleanGeneratedThemesFiles'], () => {
   return gulp.src('./bin/{js,image,css}/**/*').pipe(gulp.dest('./docs/theme/assets/gen'));
 });
 
 gulp.task('buildPlayground', shell.task(['node node_modules/webpack/bin/webpack.js --config ./webpack.playground.config.js']));
 
-function copyFile(source, target, cb) {
-  var cbCalled = false;
+gulp.task('testDoc', () => {
+  const docgenJson = JSON.parse(fs.readFileSync(docgenJsonPath));
 
-  var rd = fs.createReadStream(source);
-  rd.on('error', function(err) {
-    done(err);
-  });
-  var wr = fs.createWriteStream(target);
-  wr.on('error', function(err) {
-    done(err);
-  });
-  wr.on('close', function(ex) {
-    done();
-  });
-  rd.pipe(wr);
+  // Check if object is defined and non-empty
```

Comment:
I would not add those kind of comment, it seems superfluous here, as the code is pretty straightforward.